testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLServletAdapterTest.java

```diff
@@ -1773,6 +1779,39 @@ public void testAdapterRoleMappings() throws Exception {
 
     }
 
+    @Test
+    public void testReloginWithAndWithoutAuthSessionCookie() {
+        assertSuccessfulLogin(salesPostServletPage, bburkeUser, testRealmSAMLPostLoginPage, "principal=bburke");
+        assertSuccessfullyLoggedIn(salesPostSigServletPage, "principal=bburke");
+        // remove session cookies for the application to force re-login against the keycloak server
+        driver.manage().deleteAllCookies();
+
+        // go to the login-actions/authenticate page to remove auth session cookies
+        authenticate.setAuthRealm(SAMLSERVLETDEMO);
+        authenticate.navigateTo();
+        waitForPageToLoad();
+        assertCurrentUrlStartsWith(authenticate);
+        assertThat(driver.getPageSource(), containsString("You are already logged in."));
```

Comment:
Minor: It's a bit better to use something like:
```
infoPage.assertCurrent();
Assert.assertEquals("You are already logged in.", infoPage.getInfo());
```
I think it is safer option regarding various browsers etc.

Reply:
Done!